docs/guides/actions.md

```diff
@@ -1,5 +1,25 @@
 # Actions
 
+::: warning
+
+It is advised to configure `predictableActionArguments: true` at the top-level of your machine config, like this:
+
+```js
+createMachine({
+  predictableActionArguments: true
+  // ...
+});
+```
+
+This flag is an opt into some fixed behaviors that will be the default in v5. With this flag:
+
+- XState will always call an action with the event directly responsible for the related transition,
+- you also automatically opt-into [`preserveActionOrder`](https://xstate.js.org/docs/guides/context.html#action-order).
+
+Please be aware that you might not able to use `state` from the `meta` argument when using this flag.
```

Comment:
Can y'all add more information how to migrate? We had trouble finding a replacement for using this. Thanks!

Reply:
Could you open a discussion that would describe your use case for using this property?

Reply:
Sure!

Reply:
@Andarist Done :) https://github.com/statelyai/xstate/discussions/3511